apps/prairielearn/src/ee/lib/contextEmbeddings.ts

```diff
@@ -141,6 +146,15 @@ export async function syncContextDocuments(client: OpenAI, authnUserId: string)
         openAiUserFromAuthn(authnUserId),
       );
     }
+
+    await queryRows(
+      sql.delete_unused_doc_chunks,
+      {
+        doc_paths: allowedFilepaths,
+        chunk_ids: files.map((doc) => doc.chunkId).concat(['']),
+      },
+      QuestionGenerationContextEmbeddingSchema,
+    );
```

Comment:
As far as I can tell, you never tested this code. This immediately deletes every example question embedding you just inserted.

I'd encourage you to add `RETURNING *` to the query and log out to the job which documents were deleted for improved visibility into this process. That would have immediately made it clear what happened here.

Reply:
fixed in 775e281.